modules/runtime/nuxeo-runtime-test/src/test/java/org/nuxeo/runtime/RuntimeInitializationTest.java

```diff
@@ -47,18 +50,111 @@ public void testContributions() throws Exception {
         // do nothing
     }
 
+    protected void checkDupe(boolean detected) {
+        List<String> errors = Framework.getRuntime().getMessageHandler().getMessages(Level.ERROR);
+        if (detected) {
+            assertEquals(List.of("Duplicate component name: service:my.comp2"), errors);
+        } else {
+            assertEquals(0, errors.size());
+        }
+    }
+
     @Test
     @Deploy("org.nuxeo.runtime.test.tests:MyComp1.xml")
     @Deploy("org.nuxeo.runtime.test.tests:MyComp2.xml")
-    @Ignore("Deactivated for now since duplicate contributions are still allowed.")
+    @Deploy("org.nuxeo.runtime.test.tests:CopyOfMyComp2.xml")
     public void testContributionsWithDuplicateComponent() throws Exception {
-        boolean success = false;
-        try {
-            hotDeployer.deploy("org.nuxeo.runtime.test.tests:CopyOfMyComp2.xml");
-            success = true;
-        } catch (AssertionError e) {
-            // OK.
-        }
-        assertFalse("An exception should have been raised.", success);
+        checkDupe(true);
+    }
+
+    @Test
+    @Deploy("org.nuxeo.runtime.test.tests:MyComp1.xml")
+    @Deploy("org.nuxeo.runtime.test.tests:MyComp2.xml")
+    @Deploy("org.nuxeo.runtime.test.tests:MyComp2.xml")
+    public void testContributionsWithDuplicateComponentSameFile() throws Exception {
```

Comment:
Seems that all the `throws Exception` in methods not using the `hotDeployer` can be removed.